server/src/main/java/io/crate/metadata/IndexReference.java

```diff
@@ -114,6 +116,31 @@ public IndexReference(int position,
         this.analyzer = analyzer;
     }
 
+    public IndexReference(ReferenceIdent ident,
+                          RowGranularity granularity,
+                          DataType<?> type,
+                          ColumnPolicy columnPolicy,
+                          IndexType indexType,
+                          boolean nullable,
+                          boolean hasDocValues,
+                          int position,
+                          Symbol defaultExpression,
+                          List<Reference> columns,
+                          String analyzer) {
+        super(ident,
+              granularity,
+              type,
+              columnPolicy,
+              indexType,
+              nullable,
+              hasDocValues,
+              position,
+              defaultExpression
+        );
+        this.columns = columns;
+        this.analyzer = analyzer;
+    }
+
```

Comment:
I already added such "all fields" ctors to Geo/Index refs in OID POC (https://github.com/crate/crate/pull/13757/commits/d0dc7e0988752d8b5043fb38c7db88952e126de5#diff-a81a3c2c388134d994ae4bfb5a850764767f24f33082380c6eef1603832db19c). Basically spot that issue, had a test failure when forgot to override `assignOid`. 

Can utilise those new ctors for all future methods bearing semantics "make copy and change one field"